src/uattributes/uattributesvalidator.rs

```diff
@@ -931,13 +932,16 @@ mod tests {
         }
     }
 
-    fn create_id_for_timestamp(created_millis_ago: u64) -> UUID {
-        let now = SystemTime::now();
-        let since_epoch_start = now.duration_since(SystemTime::UNIX_EPOCH).unwrap();
-        let now_millis_since_epoch_start = u64::try_from(since_epoch_start.as_millis()).unwrap();
-
-        let creation_time = now_millis_since_epoch_start - created_millis_ago;
-        UUIDBuilder::new().build_with_instant(creation_time)
+    fn uuid_n_ms_in_past(n_ms_in_past: u64) -> UUID {
```

Comment:
Maybe we can add this as `pub(crate) fn UUIDBuilder::build_with_instant(timestap) -> UUID` so that it can (potentially) be reused in other test cases of the up-rust crate?

Reply:
I could see adding a `pub(crate)` function which does what I wrote here -- generates a UUID and then subtracts a certain amount of time from it.

I ran into issues with passing in an absolute timestamp. Read on to hear of my woes :point_down: 

--------

I was running into a fair amount of tricky debugging and headscratching before I moved away from the two-step process that was here of `build()` -> `build_with_timestamp()`, to calculating the timestamp purely in the loop of the compare-and-swap and not passing in the timestamp.

Imagine on thread 1 we had entered `build_internal()` from `build_with_timestamp()` with a timestamp of n and on thread 2 we had entered `build_internal()` from `build_with_timestamp() with a timestamp of n+1, and the previous time in the `msb` was n.

We make it to thread 2 first, see the `msb` time of n which differs from the passed in timestamp of n+1 and set the timestamp to n+1.

We hit thread 1 second, see the `msb` time of n+1 which differs from the passed in timestamp of n and set the timestamp to n (going back in time!).

If I missed the point though, please feel free to let me know your thoughts :slightly_smiling_face:

Reply:
> I could see adding a pub(crate) function which does what I wrote here -- generates a UUID and then subtracts a certain amount of time from it.

That is what I had in mind :-) Just didn't want to bury it in the test cases ...

Reply:
:laughing: Can you tell from my paragraph that I had gone through some trauma...

Sounds good, I'll move this over into `UUIDBuilder` so it can be accessed from elsewhere for other tests :+1:

Reply:
Okey dokey -- can you take another look?